### PR TITLE
Addition of filter UI and functionality in the Facade Service Layer.

### PR DESCRIPTION
**Pull request by JulianYoungEviden**
This includes extensive test-data, Error handling, logging and a working Error page.

### CODE REVIEW

File: src/EPR.RegulatorService.Frontend.Web/ViewComponents/ExpiredRegistrationSubmissionListViewComponent.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+using System.Diagnostics.CodeAnalysis;
```
Comment (shehzad-ismail8) [inline src/EPR.RegulatorService.Frontend.Web/ViewComponents/ExpiredRegistrationSubmissionListViewComponent.cs:1]: Please remove if not used

File: src/EPR.RegulatorService.Frontend.Web/ViewModels/RegistrationSubmissions/RegistrationSubmissionsListViewModel.cs [hunk 2]
```diff
@@ -11,7 +13,28 @@ public class RegistrationSubmissionsListViewModel
 
     public PaginationNavigationModel PaginationNavigationModel { get; set; }
 
-    public RegistrationSubmissionsListViewModel()
+    public RegistrationSubmissionsFilterViewModel RegistrationsFilterModel { get; set; } = new RegistrationSubmissionsFilterViewModel();
+
+    public RegistrationSubmissionsFilterModel CreateFilterModel(int? desiredPage) => new RegistrationSubmissionsFilterModel
```
Comment (shehzad-ismail8) [inline src/EPR.RegulatorService.Frontend.Web/ViewModels/RegistrationSubmissions/RegistrationSubmissionsListViewModel.cs:18]: Remove this if not needed

**Review by shehzad-ismail8 — CHANGES_REQUESTED** [review]
A couple of things to remove as necessary

**Review by johnfitzgeraldEviden — APPROVED** [review]
Approved suject to changes being completed from review

**Review by johnfitzgeraldEviden — APPROVED** [review]

**Review by shehzad-ismail8 — APPROVED** [review]

**Pull request merged**